core/adapters/eth_tx.go

```diff
@@ -105,7 +105,14 @@ func (e *EthTx) insertEthTx(input models.RunInput, store *strpkg.Store) models.R
 		err                    error
 	)
 	if e.ABIEncoding != nil {
-		txData, err = getTxDataUsingABIEncoding(e, input.Data())
+		// The requestID is present as the first element of DataPrefix (from the oracle event log).
+		// We prepend it as a magic first argument of for the consumer contract.
```

Comment:
Every time I think I understand why we have "DataPrefix" at all, I then forget.

Can you remind me what this is for?

Reply:
yah its complicated and I didn't fully understand until this PR. The dataPrefix is a set of event fields from OracleRequest, specifically 
```   
    bytes32 requestId,
    uint256 payment,
    address callbackAddr,
    bytes4 callbackFunctionId,
    uint256 cancelExpiration,
    uint256 dataVersion,
```
We save those into the input [here](https://github.com/smartcontractkit/chainlink/blob/08caa9f7de4ce0cd251d6e2c2793ec56fed452d3/core/store/models/log_events.go#L447) and then re-use them to make the fulfillOracleRequest2 call.

Reply:
Right. In the pipeline version of this I'm going to have those fields be decoded into structured data then re-encoded at the end, it makes it all much easier to reason about.

Reply:
yeah makes sense